modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java

```diff
@@ -51,6 +52,33 @@ public class ServiceConfigurationApi {
     @Resource
     private ServiceConfigurationManager serviceConfigurationManager;
 
+    /**
+     * Query the service's current configuration by id of the deployed service.
+     *
+     * @param serviceId id of the deployed service.
+     * @return ServiceConfigurationEntity.
+     */
+    @Tag(name = "ServiceConfiguration",
+            description = "APIs for managing service's configuration.")
+    @GetMapping(value = "/service/current/config/{serviceId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "Query the service's current configuration by"
+            + " id of the deployed service.")
+    @AuditApiRequest(methodName = "getCspFromServiceId")
+    public ServiceConfigurationEntity getCurrentConfigurationOfService(
```

Comment:
we must not return the entity as it is. If we do so, the deploy service entity is also returned.  This must be suppressed.